includes/class-newspack-newsletters.php

```diff
@@ -689,24 +689,31 @@ public static function api_test_mailchimp_campaign( $request ) {
 	 * @param string $id post ID.
 	 */
 	public static function retrieve_data( $id ) {
-		$mc_campaign_id      = get_post_meta( $id, 'mc_campaign_id', true );
-		$mc                  = new Mailchimp( self::mailchimp_api_key() );
-		$campaign            = $mc_campaign_id ? $mc->get( "campaigns/$mc_campaign_id" ) : null;
-		$list_id             = $campaign && isset( $campaign['recipients']['list_id'] ) ? $campaign['recipients']['list_id'] : null;
-		$interest_categories = $list_id ? $mc->get( "lists/$list_id/interest-categories" ) : null;
-		if ( $interest_categories && count( $interest_categories['categories'] ) ) {
-			foreach ( $interest_categories['categories'] as &$category ) {
-				$category_id           = $category['id'];
-				$category['interests'] = $mc->get( "lists/$list_id/interest-categories/$category_id/interests" );
+		try {
+			$mc_campaign_id      = get_post_meta( $id, 'mc_campaign_id', true );
+			$mc                  = new Mailchimp( self::mailchimp_api_key() );
+			$campaign            = $mc_campaign_id ? self::validate_mailchimp_operation( $mc->get( "campaigns/$mc_campaign_id" ) ) : null;
```

Comment:
Is there any case in which we want to return empty campaign data? Shouldn't an error be thrown if there's no `$mc_campaign_id`?

Reply:
Good idea. I had thought there was a case before the post is saved but this never occurs. Fixed inhttps://github.com/Automattic/newspack-newsletters/pull/120/commits/c56f13c702bf270b7afe1f3345c5c00fa082568a

Reply:
Still that line needs changes ;) 
```suggestion
			$campaign            = self::validate_mailchimp_operation( $mc->get( "campaigns/$mc_campaign_id" ) );
```

Reply:
Fixed in https://github.com/Automattic/newspack-newsletters/pull/120/commits/57c8ec06d7e3faf8a123160a928eef2adc9554ae